### ISSUE TITLE
SequentialLR aliases lr Tensor with initial_lr, corrupting base_lrs for chained schedulers

### ISSUE THREAD

**Issue by filipviz** [pytorch/pytorch#162359]
### 🐛 Describe the bug

1. When we use an optimizer with a Tensor learning rate (as [recommended by the docs to avoid recompilation](https://docs.pytorch.org/tutorials/recipes/compiling_optimizer_lr_scheduler.html)), `SequentialLR.__init__` assigns `group["lr"] = group["initial_lr"]`, aliasing the tensors.
2. The initial step of the first scheduler writes the lr in-place, which also mutates the `initial_lr` and all chained schedulers' `base_lrs` that reference it.
3. Chained schedulers which rely on `base_lrs` therefore start from the wrong base (e.g. scaled by `start_factor` if we used a `LinearLR` first).

For example, if we use a Tensor learning rate and `SequentialLR` with `schedulers=[LinearLR, CosineAnnealingLR]`, the `base_lrs` of our chained schedulers (and thus our `CosineAnnealingLR` outputs) are scaled by the `LinearLR`'s `start_factor`:

<img width="555" height="418" alt="Image" src="https://github.com/user-attachments/assets/cc7929ee-a4c5-4ede-9228-5eaa43dfebc9" />

`LinearLR.get_lr` does not rely on `base_lrs`, hence the jump at the milestone. With the bug fixed, we obtain the intended schedule:

<img width="555" height="415" alt="Image" src="https://github.com/user-attachments/assets/fed9db38-1f6a-43b7-be38-402f85313d42" />

Here's a reproduction:
```python
import torch
from torch import optim
import matplotlib.pyplot as plt

# 1. Use a tensor learning rate. The choice of optimizer doesn't matter.
lr = 1.0
x = torch.tensor(0.0, device='meta')
opt = optim.AdamW([x], lr=torch.tensor(lr))

# 2. Initialize our chained schedulers.
milestone, total_steps = 40, 100
start_factor, end_factor = 0.2, 1.0
warmup = optim.lr_scheduler.LinearLR(opt, start_factor, end_factor, total_iters=milestone)
decay = optim.lr_scheduler.CosineAnnealingLR(opt, T_max=total_steps-milestone)

# Note that each scheduler's base_lr aliases with the optimizer's initial_lr.
assert warmup.base_lrs[0].is_set_to(opt.param_groups[0]['initial_lr'])
assert decay.base_lrs[0].is_set_to(opt.param_groups[0]['initial_lr'])

# 3. Initialize our SequentialLR.
scheduler = optim.lr_scheduler.SequentialLR(opt, schedulers=[warmup, decay], milestones=[milestone])

# SequentialLR.__init__ aliases our optimizer's lr and initial_lr.
# for group in self.optimizer.param_groups:
#     group["lr"] = group["initial_lr"]
assert opt.param_groups[0]['lr'].is_set_to(opt.param_groups[0]['initial_lr'])

# Which means they're also aliased with the base_lrs of each scheduler!
assert (warmup.base_lrs[0].data_ptr() 
    == decay.base_lrs[0].data_ptr()
    == opt.param_groups[0]['initial_lr'].data_ptr() 
    == opt.param_groups[0]['lr'].data_ptr())

# It also calls _initial_step() on our LinearLR, setting all tensors to start_factor * lr.
assert (start_factor * lr
    == opt.param_groups[0]['lr']
    == opt.param_groups[0]['initial_lr']
    == warmup.base_lrs[0]
    == decay.base_lrs[0])

# 4. Now if we do something which breaks the aliases, e.g. save/load the optimizer's state_dict...
opt.load_state_dict(opt.state_dict())
assert not warmup.base_lrs[0].data_ptr() == opt.param_groups[0]['initial_lr'].data_ptr()

# Our base_lrs are incorrect. The LinearLR will work as intended (since get_lr updates based on the
# previous iteration, not base_lr), but the outputs of further schedulers are incorrectly scaled by 
# start_factor. In our case, these are the outputs of CosineAnnealingLR which appear after the milestone.
lrs = []
for step in range(total_steps):
    scheduler.step()
    lrs.append(scheduler.get_last_lr()[0].item())

plt.plot(lrs)
plt.show()
```

### Versions

PyTorch version: 2.8.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 15.6.1 (arm64)
GCC version: Could not collect
Clang version: 17.0.0 (clang-1700.0.13.5)
CMake version: version 4.1.1
Libc version: N/A

Python version: 3.13.7 (main, Aug 14 2025, 11:12:11) [Clang 17.0.0 (clang-1700.0.13.3)] (64-bit runtime)
Python platform: macOS-15.6.1-arm64-arm-64bit-Mach-O
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M3 Pro

Versions of relevant libraries:
[pip3] Could not collect
[conda] No relevant packages

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar
